File: Java/SpaceVIL/spacevil/src/main/java/com/spvessel/spacevil/TextInputProcessor.java
package com.spvessel.spacevil;

import com.spvessel.spacevil.Core.TextInputArgs;
import com.spvessel.spacevil.Flags.KeyMods;

final class TextInputProcessor {

    private TextInputArgs _tiargs;
    private CommonProcessor _commonProcessor;

    TextInputProcessor(CommonProcessor processor) {
        _commonProcessor = processor;
        _tiargs = new TextInputArgs();
        _tiargs.clear();
    }

    void process(long wnd, int character, int mods) {
        _tiargs.character = character;
        _tiargs.mods = KeyMods.getEnums(mods);

        if (_commonProcessor.focusedItem != null) {
            if (_commonProcessor.focusedItem.eventTextInput != null)
                _commonProcessor.focusedItem.eventTextInput.execute(_commonProcessor.focusedItem, _tiargs);
        }
    }
}

File: SpaceVIL/Engine/TextInputProcessor.cs
using System;
using Glfw3;
using SpaceVIL.Core;

namespace SpaceVIL
{
    internal class TextInputProcessor
    {
        private TextInputArgs _tiargs;
        CommonProcessor _commonProcessor;
        internal TextInputProcessor(CommonProcessor commonProcessor)
        {
            _commonProcessor = commonProcessor;
            _tiargs = new TextInputArgs();
            _tiargs.Clear();
        }

        internal void Process(Int64 wnd, uint character, KeyMods mods)
        {
            _tiargs.Character = character;
            _tiargs.Mods = mods;
            _commonProcessor.FocusedItem?.EventTextInput?.Invoke(_commonProcessor.FocusedItem, _tiargs);
        }
    }
}